### PR TITLE
Fix bootloader hash debug logging (IDFGH-5810)

### PR DESCRIPTION
**Pull request by boarchuz**
1. Fix `BOOT_LOG_LEVEL >= LOG_LEVEL_DEBUG` condition in bootloader_debug_buffer. Neither of these are defined.
2. String literals passed to bootloader_debug_buffer are linked even though it ultimately does nothing with them at lower log levels. Currently it's only used in esp_image_format.c, so I've added a static function in there to give the build system a chance to weed them out.

### CODE REVIEW

File: components/bootloader_support/src/bootloader_utility.c [hunk 1]
```diff
@@ -858,7 +858,7 @@ esp_err_t bootloader_sha256_hex_to_str(char *out_str, const uint8_t *in_array_he
 
 void bootloader_debug_buffer(const void *buffer, size_t length, const char *label)
 {
-#if BOOT_LOG_LEVEL >= LOG_LEVEL_DEBUG
+#if LOG_LOCAL_LEVEL >= 4
```
Comment (negativekelvin) [inline components/bootloader_support/src/bootloader_utility.c:861]: >= ESP_LOG_DEBUG

File: components/bootloader_support/src/esp_image_format.c [hunk 2]
```diff
@@ -792,6 +792,13 @@ static esp_err_t process_checksum(bootloader_sha256_handle_t sha_handle, uint32_
     return err;
 }
 
+static void log_debug_hash(const char *label, const uint8_t *buf)
+{
+#if LOG_LOCAL_LEVEL >= 4
```
Comment (negativekelvin) [inline components/bootloader_support/src/esp_image_format.c:797]: \>= ESP_LOG_DEBUG

**Comment by Alvin1Zhang** [comment]
Thanks for your contribution and reviewing.

**Comment by gustavonihei** [comment]
I ended up committing a similar patch while working on another component.
https://github.com/espressif/esp-idf/commit/f75f74ac40e1e7342d4ac32a43ad4f711ab75896 fixes the issue.
@Alvin1Zhang, could you please close the PR?

**Comment by boarchuz** [comment]
Thanks @gustavonihei

**Comment by Alvin1Zhang** [comment]
@boarchuz Thanks for your contribution again.

**Pull request closed without merging**